portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,13 +46,21 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
     String comment = req.getParameter("comment");
     if (comment != null && !comment.equals("")) {
       // do nothing if the comment field is empty
-      comments.add(comment);
+      Entity commentEntity = new Entity("comment");
+      commentEntity.setProperty("value", comment);
+      datastore.put(commentEntity);
     }
     resp.sendRedirect("/index.html");
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("comment");
+    PreparedQuery results = datastore.prepare(query);
+    ArrayList<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      comments.add((String)entity.getProperty("value"));
+    }
     response.setContentType("application/json;");
     response.getWriter().println((new Gson()).toJson(comments));
```

Comment:
nit: the parens around new Gson() are unnecessary - confused me for a second

Reply:
Will fix